src/intersections/lines.jl

```diff
@@ -9,20 +9,24 @@ The intersection type can be one of three types:
 2. overlap at more than one point
 3. do not overlap nor intersect
 =#
-function intersection(f, l1::Line{3,T}, l2::Line{3,T}) where {T}
+function intersection(f::Function, l1::Line{3,T}, l2::Line{3,T}) where {T}
```

Comment:
We don't need to explicitly request a `Funcion`, we can accept any `Base.Callable`. We only enforce types in the arguments when it helps with multiple dispatch.